client/client.go

```diff
@@ -67,63 +63,43 @@ func (c *Client) CompletedObjectives() <-chan protocols.ObjectiveId {
 
 // CreateVirtualChannel creates a virtual channel with the counterParty using ledger channels with the intermediary.
 func (c *Client) CreateVirtualChannel(counterParty types.Address, intermediary types.Address, appDefinition types.Address, appData types.Bytes, outcome outcome.Exit, challengeDuration *types.Uint256) protocols.ObjectiveId {
-	right, ok := (*c.store).GetTwoPartyLedger(*c.Address, intermediary)
-
-	if !ok {
-		// TODO: We need to implement proper API error handling
-		panic(fmt.Sprintf("Could not find ledger channel for participants %v,%v", *c.Address, intermediary))
+	objectiveRequest := virtualfund.ObjectiveRequest{
+		MyAddress:         *c.Address,
+		CounterParty:      counterParty,
+		Intermediary:      intermediary,
+		AppDefinition:     appDefinition,
+		AppData:           appData,
+		Outcome:           outcome,
+		ChallengeDuration: challengeDuration,
+		Nonce:             rand.Int63(), // TODO: Proper nonce management!
```

Comment:
An alternative is to expect the API to provide a Nonce. IMO, this is better than leaving a TODO, which in my experience is easy to forget about.

Having the client API require the nonce means:
- if it's a bad design, we will hear complaints about it
- users can actually use the nonce as an idempotency key

Reply:
That's good point about TODOs being easy to forget!

I've created an [issue](https://github.com/statechannels/go-nitro/issues/348) for this so we can hopefully keep track of it. I think it might make sense to tackle this outside of this PR? Adding nonces to the API seems like a big enough decision that it deserves its own PR/discussion.